README: update installation & running details

shaunanoordin opened the pull request:
## PR Overview

This PR updates the README with additional information on how to run PFE. There were some deficiencies in information that weren't uncovered until I walked our new intern through the process of setting PFE up.

- New: added explicit instructions on how to view the website (i.e. open local.zooniverse.org:3735)
- New: added important installation notes from #6155

### Status

Ready for review. Assigning to MaysaB, but everyone is free to jump in.

MaysaB commented on README.md:
```diff
@@ -30,6 +30,14 @@ Make sure you have Node 8 and `npm` 5 or greater. It's recommended you manage yo
+
+Open your web browser of choice and go to `https://localhost:3735/`
+
+If you want to _login_ via the Panoptes API and _view authenticated pages,_ then you'll need to set up and use `https://local.zooniverse.org:3735` instead of using localhost:3735. Otherwise, you'll run into CORS errors. (You need to add the hostname to your hosts file, pointing to local. Instructions are on [our Stackoverflow](https://stackoverflow.com/c/zooniverse/questions/109).)
```
This works great! I am so not getting the overlay with the errors when I first run the website - Jim's clean install suggestion is a great idea. Would it be helpful to add instructions to bypass Googles connection error page?

shaunanoordin replied:
👍

shaunanoordin commented on README.md:
```diff
@@ -30,6 +30,14 @@ Make sure you have Node 8 and `npm` 5 or greater. It's recommended you manage yo
 
 - `npm start` builds and runs the site locally.
 
+⚠️ **Note:** as of Node 16.15, running _npm install_ results in errors such as _npm ERR! ERESOLVE could not resolve_ and _Conflicting peer dependency: foobar@x.y.z_ You can bypass this problem by instead running `npm install --legacy-peer-deps`. Please see [issue 6155](https://github.com/zooniverse/Panoptes-Front-End/issues/6155) for more details.
```
👍

shaunanoordin commented:
Thanks both for the feedback!

## PR Update

- Changed: instructions now favour `npm ci` instead of `npm install`
- Added: section explaining why we prefer `npm ci`
- Added: section on troubleshooting web browsers blocking local websites

eatyourgreens commented on README.md:
```diff
@@ -60,7 +84,7 @@ While editing, do your best to follow style and architecture conventions already
 ### What to do if it doesn't run
 
-Try `rm -rf ./node_modules && npm install` to freshen up your dependencies. And read the warnings, they should tell you if you're using the wrong version of Node or npm or if you're missing any dependencies. If you use `docker-compose` to build and test the site, you shouldn't run into any problems with the Node version, but `docker-compose build` will build a new image with a fresh `npm install` ...
+Try `rm -rf ./node_modules && npm ci` to freshen up your dependencies. And read the warnings, they should tell you if you're using the wrong version of Node or npm or if you're missing any dependencies. If you use `docker-compose` to build and test the site, you shouldn't run into any problems with the Node version, but `docker-compose build` will build a new image with a fresh `npm ci`.
```
`npm ci` deletes any installed `node_modules` so `rm -rf ./node_modules` is redundant here.
```suggestion
Try `npm ci` to freshen up your dependencies. And read the warnings, they should tell you if you're using the wrong version of Node or npm or if you're missing any dependencies. If you use `docker-compose` to build and test the site, you shouldn't run into any problems with the Node version, but `docker-compose build` will build a new image with a fresh `npm ci`.
```

MaysaB reviewed (approved):
LGTM!
